src/js/flows/renameSpace.js

```diff
@@ -0,0 +1,23 @@
+/* @flow */
+import type {Thunk} from "../state/types"
+import Spaces from "../state/Spaces"
+import Cluster from "../state/Clusters"
+import Tabs from "../state/Tabs"
+import Search from "../state/Search"
+
+export default (clusterId: string, spaceId: string, name: string): Thunk => (
+  dispatch,
+  getState
+) => {
+  const state = getState()
+  const zClient = Cluster.getZealot(clusterId)(state)
+  const tabs = Tabs.getData(state)
+
+  zClient.spaces.update(spaceId, {name}).then(() => {
+    dispatch(Spaces.rename(clusterId, spaceId, name))
+    tabs.forEach((t) => {
```

Comment:
Maybe this can be removed one day in the future? And we can refactor Search to only need the space id?